employees/tests.py

```diff
@@ -288,142 +290,302 @@ def test_should_month_update_view_work(self):
+    # reason- loading fikstura_cities takes very long time
+
+    def setUp(self):
+        self.client.login(username='pawel.kisielewicz@codepoets.it', password='codepoets')
+        zbigniew_adamski = Employee.objects.get(id=3)
+        assert ('Zbigniew Adamski' == str(
+            zbigniew_adamski)), 'Zbigniew Adamski should be in fixtures, check if file contains Zbigniew Adamski or problem with loading fixtures'
+
+    def test_should_business_logic_extension_work_properly(self):
+        dict = {2: [True, '2018-10-19', 'Jakub Czachura'],
+                3: [True, '2018-01-19', 'Zbigniew Adamski'],
+                4: [True, '2017-02-27', 'Czarny Lodziarz'],
+                6: [True, '2018-01-19', 'Andrzej Strzelba'],
+                12: [True, '2017-03-02', 'Ccc Ccc'],
+                11: [False, 'null', 'Bbb Bbb'],
+                29: [False, 'null', 'Paweł Testowy'],
+                }
+
+        for key in dict.keys():
+            self.__should_contract_extension_work_properly(key, dict.get(key)[0], dict.get(key)[1], dict.get(key)[2],
+                                                           'add_1_id')
+            # self.__should_contract_extension_work_properly(key, dict.get(key), 'add_3_month')
+
+    def __should_contract_extension_work_properly(self, id, answer_status, expected_date, expected_name, extension):
+        kwargs = {'HTTP_X_REQUESTED_WITH': 'XMLHttpRequest'}
+        url = reverse('employee_action')
+
+        get_data = {'extensionLength': extension,
+                    'employeeId': id,
+                    }
+
+        response = self.client.post(url, get_data, **kwargs)
+        self.assertEqual(response.status_code, 200)
 
-    def test_should_hide_paid_employees_filter_work_properly(self):
-        admin_login(self)
-        url = reverse_lazy('employees')
-        unpaid_response = self.c.get(url + "?hide_paid_employees_filter=on")
-        self.assertQuerysetEqual(
-            unpaid_response.context['all_employee_list'],
-            ['<Employee: Czarny Lodziarz>']
-        )
+        import json
+        dict = json.loads(response.content)
 
-    def test_should_multiple_filters_at_a_time_work_properly(self):
-        admin_login(self)
-        url = reverse_lazy('employees')
-        response = self.c.get(url + "?employee_filter=w&position_filter=Sale&position_filter=Other"
-                                    "&position_filter=Production&hide_zero_salary_months=on")
-        self.assertQuerysetEqual(
-            response.context['all_employee_list'],
-            ['<Employee: Jarosław K.>', '<Employee: Waldemar Kiepski>']
-        )
+        status = dict[0]
+        name = dict[1]
+        date = dict[2]
+
+        self.assertEqual(status, answer_status)
+        self.assertEqual(name, expected_name)
+        self.assertIn(expected_date, date)
 
-class MonthFilterTests(TestCase):
 
+@skip
+class MonthFilterTests(TestCase):
```

Comment:
Why is it skipped? Can it be fixed? If not maybe the whole test should be dropped.

Reply:
fixed, it was your old tests. Some are rewritten now, redundant deleted